### ISSUE TITLE
Question: Using functor to store position information

### ISSUE THREAD

**Issue by Centril** [BNFC/bnfc#176]
According to `bnfc --help`:
`--functor       Make the AST a functor and use it to store the position of the nodes`

How would you go about storing the position of the nodes?
The type that

``` haskell
parseProgram = pProgram . myLexer
```

yields is:

``` haskell
parseProgram :: String -> Err (Program ())
```

I.e: no position information is added... So does that mean that you have to use something else than `happy`, or can you get it to add that position information for you?

**Comment by pascalh** [comment]
If I read the bnfc sources correctly source positions are not supported by the happy backend of bnfc. Actually the lexer generated by alex yields a list of tokens which are annotated with their source positions, but the happy parser ignores them.

**Comment by gdetrez** [comment]
@pascalh is correct, the help comment describe the intended behavior but I still need to find time to complete the feature. The functor part works but not the position part...

Sorry for the inconsistent documentation :confused:.

**Comment by Centril** [comment]
Any updates on this @gdetrez ?

**Comment by gdetrez** [comment]
@Centril partly: I have done some work on this but I was interrupted by other things...

I'll try to rebase and push what I already have this week and see if I can finish it soon-ish. Would you be able to provide some feedback?

**Comment by Centril** [comment]
@gdetrez Sure, what type of feedback do you want? Peer review or testing (or both)?

**Comment by JKTKops** [comment]
Any news on this? This would be a very useful feature!

**Comment by andreasabel** [comment]
Seems like this feature sits on an unmerged branch https://github.com/BNFC/bnfc/compare/176-source-position which my predecessor left behind.  After three years of course it does not merge cleanly any more.  I don't know if there was a reason this was not finished.

I'd be happy for someone to pick it up and integrate it with the current source of BNFC.

**Comment by andreasabel** [comment]
This has been implemented by @Commelina in #327 and will be released with 2.9.1.

**Comment by andreasabel** [comment]
Released in 2.9.1.
